webapp/src/Service/SubmissionService.php

```diff
@@ -530,8 +530,9 @@ public function submitSolution(
             if ($source !== 'shadowing' && $language->getFilterCompilerFiles()) {
                 $matchesExtension = false;
                 foreach ($language->getExtensions() as $extension) {
-                    $extensionLength = strlen($extension);
-                    if (substr($file->getClientOriginalName(), -$extensionLength) === $extension) {
+                    $extensionWithDot = '.' . ltrim($extension, '.');
```

Comment:
Why is the ltrim needed?

Reply:
I was scared of this change backfiring.

Let's say an end user has added a new language, say `BQN`. He adds the extension `.bqn` to the extension list (accidentally with the dot) and turns on the filter. This works fine with the old code, and teams submitting BQN files get through.

The server gets updated and includes this commit. Now when users try to submit for this Language, they get the error
```None of the submitted files match... (allowed: .bqn)```.
Which does not help at all.

`extensionWithDot` would now contain two dots, which does not match any filename. The `ltrim` ensures that the variable always starts with only one dot and avoids weird errors.

What do you think? Should this be clarified/handled elsewhere?

Reply:
Since there is several pieces of code that work with the configured filename extensions, I think I would prefer to make the source format of our extensions simple/predictable, instead of handling it with or without `.` on all code that interprets it.

Although it does not explicitly forbid it, the [CLICS specification](https://ccs-specs.icpc.io/2023-06/contest_api#languages) only provides extensions bare, i.e. without `.`.

So it's probably best, in my opinion, to start forbidding adding `.` in the admin interface for a language, and assume in this code it will not already be present.

Reply:
Good point.